File: java programs/DSA/nCr.java
public class nCr {
    public static void main(String args[]) {
        int n = 6;
        int r = 2;
        int factN = 1;
        for (int i = 1; i <= n; i++) {
            factN = factN * i;
        }
        int factR = 1;
        for (int i = 1; i <= r; i++) {
            factR = factR * i;
        }
        int factNR = 1;
        for (int i = 1; i <= n - r; i++) {
            factNR = factNR * i;
        }
        System.out.println(factN);
        System.out.println(factR);
        System.out.print(factNR);
        double den = (double) factR * (double) factNR;
        double num = (double) factN;
        double res = num / den;
        System.out.println(res);
    }
}

File: c programs/nCr.c
#include<stdio.h>
int fact(int n)
{
    if(n==0)
    {
        return 1;
    }
    return fact(n-1)*n;
}
int nCr(int n,int r)
{
    int num,den;
    num=fact(n);
    den=fact(r)*fact(n-r);
    return num/den;
}
int ncr(int n,int r)
{
    if(r==0||n==r)
    {
        return 1;
    }
    return ncr(n-1,r-1)+(n,r-1);
}
void main()
{
printf("%d \n",nCr(5,3));
printf("%d \n",ncr(5,3));
}